android/titanium/src/java/org/appcelerator/titanium/TiBlob.java

```diff
@@ -872,6 +872,60 @@ public TiBlob imageAsCompressed(Number compressionQuality)
 		return result;
 	}
 
+	@Kroll.method
+	public TiBlob imageAsWebP(Number compressionQuality)
+	{
+		Bitmap img = getImage();
+		if (img == null) {
+			return null;
+		}
+
+		float quality = 1f;
+		if (compressionQuality != null) {
+			quality = compressionQuality.floatValue();
+		}
+
+		TiBlob result = null;
+		ByteArrayOutputStream bos;
+
+		try {
+			bos = new ByteArrayOutputStream();
+			if (img.compress(
+				quality == 1f ? CompressFormat.WEBP_LOSSLESS : CompressFormat.WEBP_LOSSY, (int) (quality * 100), bos)
+			) {
+				byte[] data = bos.toByteArray();
+
+				BitmapFactory.Options bfOptions = new BitmapFactory.Options();
+				bfOptions.inPurgeable = true;
+				bfOptions.inInputShareable = true;
+
+				result = TiBlob.blobFromData(data, "image/webp");
+			}
+		} catch (OutOfMemoryError e) {
+			TiBlobLruCache.getInstance().evictAll();
+			Log.e(TAG, "Unable to get the thumbnail image. Not enough memory: " + e.getMessage(), e);
```

Comment:
The error messages are misleading, you're not creating a thumbnail image - you're converting to a WebP image.